cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -112,6 +133,110 @@ void kafka_consumer::consume_to_buffer()
+  for (std::list<std::string>::iterator it = dump->begin(); it != dump->end();) {
+    key = (*it);
+    it++;
+    val = (*it);
+    it++;
+    configs.insert(std::pair<std::string, std::string>{key, val});
+  }
+  return configs;
+}
+
+int64_t kafka_consumer::get_committed_offset(std::string topic, int partition)
+{
+  std::vector<RdKafka::TopicPartition *> toppar_list;
+  // toppar_list.push_back(find_toppar(topic, partition));
+  toppar_list.push_back(RdKafka::TopicPartition::create(topic, partition));
+
+  // Query Kafka to populate the TopicPartitions with the desired offsets
+  RdKafka::ErrorCode err = consumer->committed(toppar_list, default_timeout);
+
+  int64_t offset = toppar_list[0]->offset();
+  if (offset > 0) {
+    return offset;
+  } else {
+    return 0;
+  }
+}
+
+std::map<std::string, int64_t> kafka_consumer::get_watermark_offset(std::string const &topic,
+                                                                    int partition,
+                                                                    int timeout,
+                                                                    bool cached)
+{
+  int64_t low;
+  int64_t high;
+  std::map<std::string, int64_t> results;
+  RdKafka::ErrorCode err;
+
+  if (cached == true) {
+    err = consumer->get_watermark_offsets(topic, partition, &low, &high);
+  } else {
+    err = consumer->query_watermark_offsets(topic, partition, &low, &high, timeout);
+  }
+
+  if (err != RdKafka::ErrorCode::ERR_NO_ERROR) {
+    if (err == RdKafka::ErrorCode::ERR__PARTITION_EOF) {
+      results.insert(std::pair<std::string, int64_t>("low", low));
+      results.insert(std::pair<std::string, int64_t>("high", high));
+    } else {
+      CUDF_FAIL("Error retrieving Kafka watermark offset from broker");
+    }
+  } else {
+    results.insert(std::pair<std::string, int64_t>("low", low));
+    results.insert(std::pair<std::string, int64_t>("high", high));
+  }
+
+  return results;
+}
+
+bool kafka_consumer::commit_offset(std::string const &topic, int partition, int64_t offset)
+{
+  std::vector<RdKafka::TopicPartition *> partitions_;
+  RdKafka::TopicPartition *toppar = RdKafka::TopicPartition::create(topic, partition, offset);
+  if (toppar != NULL) {
+    toppar->set_offset(offset);
+    partitions_.push_back(toppar);
+    RdKafka::ErrorCode err = consumer->commitSync(partitions_);
+    return true;
+  } else {
+    return false;
+  }
+}
+
+bool kafka_consumer::unsubscribe()
+{
+  RdKafka::ErrorCode err = consumer.get()->unassign();
+  if (err != RdKafka::ERR_NO_ERROR) {
+    printf(
+      "Timeout occurred while unsubscribing from Kafka Consumer "
+      "assignments.\n");
+    return false;
+  } else {
+    return true;
+  }
+}
+
+bool kafka_consumer::close(int timeout)
+{
+  RdKafka::ErrorCode err = consumer.get()->close();
+  if (err != RdKafka::ERR_NO_ERROR) {
+    printf("Timeout occurred while closing Kafka Consumer\n");
+    return false;
+  } else {
+    return true;
+  }
```

Comment:
I like to use the pattern as such:
```
  if (err != RdKafka::ERR_NO_ERROR) {
    printf("Timeout occurred while closing Kafka Consumer\n");
    return false;
  }
  return true;
```
There is no need for the `else` and it would've made the abandoned remaining lines more obvious.